src/unstract/sdk/file_storage/fs_permanent.py

```diff
@@ -75,7 +82,10 @@ def read(
         """
         try:
             # Lazy copy to the destination/remote file system
-            self._copy_on_write(path)
+            if legacy_storage_path:
+                self._copy_on_read(path, legacy_storage_path)
             return super().read(path, mode, encoding, seek_position, length)
+        except FileNotFoundError:
+            logger.warning(f"File {path} not found. Ignoring.")
```

Comment:
NIT: I suggest adding context on what we're trying to do. Are we ignoring copying a file to remote storage here?
As long as this log makes sense when we have all the other logs, I'm fine with it
`File /home/expected_file.txt not found. Ignoring.`

Also considering adding quotes to make it more readable

Reply:
Will pick it up in the next PR.